### ISSUE TITLE
[gatsby-plugin-preload-fonts]font flashes working with gobal sass style in production mode

### ISSUE THREAD

**Issue by thundermiracle** [gatsbyjs/gatsby#26631]
## Description

Use `gatsby-plugin-sass` to inject global style and use `gatsby-plugin-preload-fonts` to pre-load fonts in production mode causes flash of unstyled content.

### Steps to reproduce

1. Import global style like this:
    ```js
    {
      resolve: "gatsby-plugin-sass",
      options: {
        data: `@import "core.scss";`,
        includePaths: [path.resolve(__dirname, "src/style")],
      },
    },
    ```
2. following gatsby-plugin-preload-fonts's steps including generate `font-preload-cache.json` and add it to gatsby-config.js

3. run `gatsby build` & `gatsby serve` and FOUT

4. check the `Network` section in chrome and found that fonts were not pre-downloaded.

5. Check the outputted `index.html`, and found that `<link as='font' ...>` part is injected **AFTER** the `<style data-href="/styles.xxxx.css">@font-face`.

As codesandbox can't run `gatsby build` & `gatsby serve`, I created a reproduce repo: 
https://github.com/thundermiracle/gatsby-starter-default

### Expected result
Perhaps inject  `<link as='font' ...>` **BEFORE** the style part will fix the issue?

### Actual result
FOUT

**Comment by muhzulzidan** [comment]
how about using typeface-fonts ? like typeface-roboto.. ? it is fairly easy to use just do ```npm i typeface-fonts``` then add it into ```gatsby-browser.js``` then use it in like usual ```font family : "Roboto", sans-serif```

**Comment by thundermiracle** [comment]
@muhzulzidan 
Yes, I'm using self-host `typeface-xxx`, but it only speed up the loading, the flash problem still exists.
So I tried the preload-fonts plugin.

**Comment by blainekasten** [comment]
It looks like you missed the part of the setup where you have to run `npm run preload-fonts` before you build. 

https://github.com/gatsbyjs/gatsby/tree/master/packages/gatsby-plugin-preload-fonts#how-to-use

**Comment by thundermiracle** [comment]
@blainekasten 
I did run `npm run preload-fonts` and also did mention it in my description.
![image](https://user-images.githubusercontent.com/25689126/91231401-21542880-e768-11ea-84e6-1f0809949f85.png)

**Comment by thundermiracle** [comment]
@blainekasten 
I don’t understand why not reopen this unresolved issue?

**Comment by mixja** [comment]
Based upon my own testing the order doesn't matter in the head (you can verify this by modified the built static content and changing the order yourself manually).  

On slow connections, you will get FOIT or FOUT depending on your `font-display` setting - there is nothing you can really do about this other than until perhaps Chrome allows a configurable loading period for the `optional` setting (see https://web.dev/preload-optional-fonts/).

Preload benefits faster connections where chances are if you preload you will have downloaded fonts before the first Contentful paint.

**Comment by thundermiracle** [comment]
@mixja 
Thanks for your response.
I was trying to switch font plugin from `gatsby-plugin-prefetch-google-fonts` to official `gatsby-plugin-preload-fonts` which caused flash.

>based upon my own testing the order doesn't matter in the head (you can verify this by modifying the built static content and changing the order yourself manually)
I had a glance at both plugins' sourcecode but didn't find out any differences. So I compared the output `index.html` and found that the only difference between them were the position of the `preload`.
After I changed the position the FOUT gone away, (In [this starter](https://github.com/thundermiracle/gatsby-startbootstrap-agency), the [example project](https://github.com/thundermiracle/gatsby-starter-default) is too lightweight to confirm the flash) so I opened this issue.

As gatsby core team member @blainekasten is not interesting in this one, I think I could only count on my own and make a private useful plugin anytime.

Really appreciate for your comment.
